R/prepInputs.R

```diff
@@ -813,13 +808,66 @@ appendChecksumsTable <- function(checkSumFilePath, filesToChecksum,
   filesInArchive <- NULL
   if (!is.null(funWArgs$fun)) {
     if (file.exists(archive[1])) {
-      filesInArchive <- funWArgs$fun(archive[1], list = TRUE)
-      if ("Name" %in% names(filesInArchive)) {
-        # for zips, rm directories (length = 0)
-        filesInArchive <-
-          filesInArchive[filesInArchive$Length != 0,]$Name
-      } else { # untar
-        filesInArchive
+      if (!file_ext(archive[1]) == "rar") {
+        filesInArchive <- funWArgs$fun(archive[1], list = TRUE)
+        if ("Name" %in% names(filesInArchive)) {
+          # for zips, rm directories (length = 0)
+          filesInArchive <-
+            filesInArchive[filesInArchive$Length != 0,]$Name
+        } else { # untar
+          filesInArchive
+        }
+      } else {
+        message(paste0("The archive is a .rar file. preProcess will try a system call of 'unrar'."))
+        wd <- getwd()
+        tempDir <- file.path(dirname(archive[1]), "extractedFiles")
+        dir.create(tempDir, showWarnings = FALSE)
+        setwd(tempDir)
+        OSystem <- Sys.info()["sysname"]
+        if (OSystem == "Windows") {
+          find7zPath <- list.files("C:/Program Files/7-Zip", pattern = "7z.exe", recursive = TRUE, full.names = TRUE)
+          if (!length(find7zPath) > 0) {
+            stop("prepInputs did not find '7-Zip' installed in 'C:/Program Files/7-Zip'.",
+                 " Please install '7-Zip' in this directory before running prepInputs for a '.rar' archive")
+          } else {
+            system(paste0("\"", find7zPath, "\"", " e -o",
+                          tempDir, " ",
+                          archive[1]),
+                   wait = TRUE, invisible = TRUE,
+                   show.output.on.console = FALSE)
+          }
+        } else {
+          hasUnrar <- system("unrar", ignore.stdout = TRUE)
```

Comment:
again, probably safer to use `Sys.which`

Reply:
you can do the system checks at the start of the function, that way, if they fail you aren't wasting time running code and can message the user to install the corresponding unrar utils.